errors.go

```diff
@@ -123,50 +128,67 @@ func wrapDecodeError(document []byte, de *decodeError) error {
 
 func formatLineNumber(line int, width int) string {
 	format := "%" + strconv.Itoa(width) + "d"
+
 	return fmt.Sprintf(format, line)
 }
 
 func linesOfContext(document []byte, highlight []byte, offset int, linesAround int) ([][]byte, [][]byte) {
+	return beforeLines(document, offset, linesAround), afterLines(document, highlight, offset, linesAround)
+}
+
+func beforeLines(document []byte, offset int, linesAround int) [][]byte {
 	var beforeLines [][]byte
 
-	// Walk the document in reverse from the highlight to find previous lines
+	// Walk the document backward from the highlight to find previous lines
 	// of context.
 	rest := document[:offset]
+backward:
 	for o := len(rest) - 1; o >= 0 && len(beforeLines) <= linesAround && len(rest) > 0; {
-		if rest[o] == '\n' {
+		switch {
```

Comment:
I prefer the `if .. else if ... else` form as it does not require labeling the `for` loop to break. Is this something that can be disabled in the linter?

Reply:
I thin it should be. The recommendation comes from the gocritic linter and is following idiomatic go https://golang.org/doc/effective_go#switch

Reply:
Given this example is called out in effective go, let's use the switch form you propose and i'll get used to it. Thanks for pointing this out!